src/simplewallet/simplewallet.cpp

```diff
@@ -3281,6 +3281,36 @@ static bool datestr_to_int(const std::string &heightstr, uint16_t &year, uint8_t
   }
   return true;
 }
+//----------------------------------------------------------------------------------------------------
+std::time_t simple_wallet::get_timestamp_from_word(std::string old_language, std::string word)
+{
+  std::string mnemonic_language = old_language;
+
+  const Language::Base *language = NULL;
+  const std::vector<const Language::Base*> language_list = crypto::ElectrumWords::get_language_list();
+  for (const Language::Base *l: language_list)
+  {
+    if (mnemonic_language == l->get_language_name() || mnemonic_language == l->get_english_language_name())
+      language = l;
+  }
+  
+  const std::vector<std::string> &word_list = language->get_word_list();
+  
+  const std::vector<std::string>::const_iterator it = std::find(word_list.begin(), word_list.end(), word);
```

Comment:
The word could contain a number right? Like abbey17 or something. This doesn't parse out the numbers and won't match. If the word doesn't exist also you get the end iterator, not sure what distance(..) on an end_iterator does but it probably produces some invalid values or worse, crashes.